lib/Serverless.js

```diff
@@ -30,8 +30,12 @@ class Serverless {
     this.variables = new Variables(this);
     this.pluginManager = new PluginManager(this);
 
+    // create a new CLI instance
+    const cli = new CLI(this);
```

Comment:
Can't we just move ` this.cli = new this.classes.CLI(this);` from L59 here?

With this proposal two CLI instances are created in all cases.

Reply:
yeah, i wasn't sure what to do about that. it's created later to allow customization, but i need to parse the config option here in order to load the config file.  Does anything use the CLI customization option?

Reply:
> It's created later to allow customization

Why creating it here (earlier) will prevent customization? As I see we're in a context of instance, so it should have same effect (?)

Reply:
Note sure if I'm following 🤔 What customization are we talking about here? The [inline-if](https://github.com/serverless/serverless/pull/6216/files#diff-55402a088565d14960a8e9e477f962bbR156) for `commandObject`?

Reply:
As i said, i have no clue what uses this ability, but it's tested for like this: https://github.com/serverless/serverless/blob/master/lib/Serverless.test.js#L140-L148

Reply:
I have little idea either. Blaming brings up this PR --> https://github.com/serverless/serverless/pull/4160

Reply:
I looked into that, and it appears that our moves are limited (as we don't want to break some eventual user customizations.)

Ideal solution (imo), would be to move out CLI args parsing to `bin/serverless` and do not keep it internally, so Serverless class directly receives (already parsed out) `commands` and `options` (it'll also make a most friendly programmatical interface).

Still that change will be breaking, so I would leave it as TODO, for some near breaking release (once we sort out our versioning).

And now, I think cleaner would be, to not initialize here whole `CLI` class, but just do `minimist(process.argv.slice(2))` and seclude out of that eventual config setting (I would also add TODO comment, that signal that we will rework our commands args parsing)

Additionally I would validate existence of  `process.join(process.cwd(),customConfig)`,
With what's proposed now, we fallback to `serverless.yml` if custom config doesn't exist, but that I think can turn quite confusing (e.g. user may make typo in name, and be puzzled why original config is loaded)

Reply:
Agreed that I don't want to introduce breaking cahnges. 

> And now, I think cleaner would be, to not initialize here whole CLI class, but just do minimist(process.argv.slice(2)) and seclude out of that eventual config setting (I would also add TODO comment, that signal that we will rework our commands args parsing)

that makes sense. i'll do the simple parsing here

> Additionally I would validate existence of process.join(process.cwd(),customConfig),
With what's proposed now, we fallback to serverless.yml if custom config doesn't exist, but that I think can turn quite confusing (e.g. user may make typo in name, and be puzzled why original config is loaded)

Good point. It actually doesn't fallback, but it breaks in a confusing way, so i'll make that error better.